src/js/components/CheckBox/CheckBox.js

```diff
@@ -195,7 +195,9 @@ const CheckBox = forwardRef(
     return (
       <StyledCheckBoxContainer
         aria-label={a11yTitle}
-        fill={fill}
+        // Note: StyledCheckBoxContainer is based on `styled.label`, which for
+        // some reason requires the `fill` prop to be of type string.
+        fill={fill && 'true'}
```

Comment:
I think this is good!

Reply:
If we're trying to avoid a property leaking into the DOM, we should rename it with something like `fillProp`. That allows StyledCheckBoxContainer to have access to the value without react automatically matching it to a DOM attribute.